dss/api/search.py

```diff
@@ -10,14 +11,12 @@
 from .. import Config, Replica, ESIndexType, dss_handler, get_logger, DSSException
 from ..util.es import ElasticsearchClient
 
-# TODO Adding replica as a search parameter and including tests for gcp
-# will be done in a different PR.
-replica = "aws"
-
 @dss_handler
-def post(json_request_body: dict):
+def post(json_request_body: dict, replica: typing.Optional[str]=None):
```

Comment:
why not make it required?

Reply:
Trent and I debated about whether it should be required or not, and looked at files.py and bundles.py to see if we could infer a convention regarding this. Some of the files and bundles operations treat the replica as optional and default to "aws" (accompanied by a todo to use the local replica when the time comes), and others require the replica option be specified. I proposed making it optional, thinking along the lines of minimizing the number of options a user needs to specify and providing reasonable default values. Of course, the option to specify a replica is available if they choose. 
The subscription route requires the replica parameter, which makes sense there. 
We are happy to make this a required parameter if you think that is a better fit with our existing conventions and future direction.

Reply:
I think for search, it should be required.

Rationale: Someone searching for something wants to search for data close to their compute.  The reason it can be optional for get bundle/file is that people are unlikely to just divine the url.  They have to search for something to get the initial url.

If it makes you feel better, I am happy to make get bundle/file require a replica for consistency's sake. :)

Reply:
This is a problem of customizability vs simplicity. The harder we make it to use the less likely it will be used. Since this is a key feature we should make it as easy to use as possible. Making replica optional with a default value simplifies life for the user. If the user want to customize and optimize they can later choose to specify the replica.

Reply:
I agree with @Bento007, replica should not be required. We should make a request geolocator that assigns the replica based on the network CIDR the request is coming from.

Reply:
Sounds like a future feature we will need to implement on data-store-cli.

Reply:
Oh, I see what you're saying.  You're saying search should have the replica field be optional?

I think it makes more sense for the front end to assign a default replica than it does for the backend.  Having the back end automatically determine the replica means the same search could yield different results to different users.